test/unit/Cardano/Wallet/CoinSelection/FeeSpec.hs

```diff
@@ -0,0 +1,333 @@
+            , fFee = 3
+            , fDust = 0
+            }) (Right $ FeeOutput
+            { csInps = [20,20,20]
+            , csOuts = [14,18,19]
+            , csChngs = [4,1]
+            })
+
+        feeUnitTest (FeeFixture
+            { fInps = [20,20,20]
+            , fOuts = [14,18,19]
+            , fChngs = [6,2,1]
+            , fExtraUtxo = []
+            , fFee = 3
+            , fDust = 1
+            }) (Right $ FeeOutput
+            { csInps = [20,20,20]
+            , csOuts = [14,18,19]
+            , csChngs = [3]
+            })
+
+        feeUnitTest (FeeFixture
+            { fInps = [20,20,20]
+            , fOuts = [14,17,19]
+            , fChngs = [6,3,1]
+            , fExtraUtxo = []
+            , fFee = 3
+            , fDust = 0
+            }) (Right $ FeeOutput
+            { csInps = [20,20,20]
+            , csOuts = [14,17,19]
+            , csChngs = [4,2]
+            })
+
+        feeUnitTest (FeeFixture
+            { fInps = [20,20,20]
+            , fOuts = [14,17,19]
+            , fChngs = [6,3,1]
+            , fExtraUtxo = []
+            , fFee = 3
+            , fDust = 1
+            }) (Right $ FeeOutput
+            { csInps = [20,20,20]
+            , csOuts = [14,17,19]
+            , csChngs = [4,2]
+            })
+
+        feeUnitTest (FeeFixture
+            { fInps = [20,20,20]
+            , fOuts = [14,17,19]
+            , fChngs = [6,3,1]
+            , fExtraUtxo = []
+            , fFee = 3
+            , fDust = 2
+            }) (Right $ FeeOutput
+            { csInps = [20,20,20]
+            , csOuts = [14,17,19]
+            , csChngs = [4]
+            })
+
+        feeUnitTest (FeeFixture
+            { fInps = [20]
+            , fOuts = [17]
+            , fChngs = [3]
+            , fExtraUtxo = []
+            , fFee = 4
+            , fDust = 0
+            }) (Left $ CannotCoverFee 1)
+
+        feeUnitTest (FeeFixture
+            { fInps = [20,20]
+            , fOuts = [16,18]
+            , fChngs = [4,2]
+            , fExtraUtxo = []
+            , fFee = 6
+            , fDust = 0
+            }) (Right $ FeeOutput
+            { csInps = [20,20]
+            , csOuts = [16,18]
+            , csChngs = []
+            })
+
+        feeUnitTest (FeeFixture
+            { fInps = [20,20]
+            , fOuts = [16,18]
+            , fChngs = [4,2]
+            , fExtraUtxo = []
+            , fFee = 6
+            , fDust = 2
+            }) (Left $ CannotCoverFee 2)
+
+        feeUnitTest (FeeFixture
+            { fInps = [10]
+            , fOuts = [7]
+            , fChngs = [3]
+            , fExtraUtxo = [1]
+            , fFee = 5
+            , fDust = 0
+            }) (Left $ CannotCoverFee 2)
+
+        feeUnitTest (FeeFixture
+            { fInps = [10]
+            , fOuts = [7]
+            , fChngs = [3]
+            , fExtraUtxo = [2]
+            , fFee = 5
+            , fDust = 0
+            }) (Left $ OutOfBoundFee 0 5)
```

Comment:
this one I do not understand. From `senderPaysFee` I have : 
```
coinSel' = CoinSelection {inputs = [(TxIn {inputId = Hash {getHash = "\170\223\131K\130@\DLE2\202\159"}, inputIx = 0},TxOut {address = Address {getAddress = "ADDR03"}, coin = Coin {getCoin = 10}}),(TxIn {inputId = Hash {getHash = "\219\CANf\153\159\&9\189\154\&9\133"}, inputIx = 0},TxOut {address = Address {getAddress = "ADDR01"}, coin = Coin {getCoin = 2}})], outputs = [TxOut {address = Address {getAddress = "ADDR01"}, coin = Coin {getCoin = 7}}], change = [Coin {getCoin = 5}]}
```
and after `computeFee coinSel'` I have Coin 0 ... The computation is : 
```
collapse : p>m : Coin {getCoin = 10} > Coin {getCoin = 7} ps:[Coin {getCoin = 2}] ms[Coin {getCoin = 5}]
collapse : p<m : Coin {getCoin = 3} < Coin {getCoin = 5} ps:[Coin {getCoin = 2}] ms[]
collapse : p==m : Coin {getCoin = 2} ps:[] ms[]
collapse : plus : []
```
I need to think about it - though it should be **Right**

Reply:
addressed in new commit